defold-spine/editor/src/spineext.clj

```diff
@@ -349,38 +353,40 @@
-        renderable-transform (Matrix4d. (:world-transform renderable)) ; make a copy so we don't alter the original
-
-        ro-matrix (doto (Matrix4d. ro-transform) (.transpose))
-        shader-world-transform (doto renderable-transform (.mul ro-matrix))
-        use-index-buffer (not= (.m_UseIndexBuffer ro) 0)
-        triangle-mode (if (not= (.m_IsTriangleStrip ro) 0) GL/GL_TRIANGLE_STRIP GL/GL_TRIANGLES)
-        render-args (merge render-args
-                           (math/derive-render-transforms shader-world-transform
-                                                          (:view render-args)
-                                                          (:projection render-args)
-                                                          (:texture render-args)))]
-    (gl/set-blend-mode gl blend-mode)
-    (shader/set-uniform shader gl "world_view_proj" (:world-view-proj render-args))
-    (when (not= (.m_SetFaceWinding ro) 0)
-      (gl/gl-front-face gl face-winding))
-    (when (not= (.m_SetStencilTest ro) 0)
-      (set-stencil-test-params! gl (.m_StencilTestParams ro)))
-    (when (not use-index-buffer)
-      (gl/gl-draw-arrays gl triangle-mode start count))
-    (when use-index-buffer
-      (gl/gl-draw-elements gl triangle-mode start count))
-    (gl/set-blend-mode gl :blend-mode-alpha)))
+       (let [start (.m_VertexStart ro)                      ; the name is from the engine, but in this case refers to the index
+             count (.m_VertexCount ro)
+             is-selection-pass? (= (:selection (:pass render-args)) true)
+             renderable-user-data (:user-data renderable)
+             renderable-blend-mode (:blend-mode renderable-user-data)
+             blend-mode (if (= renderable-blend-mode :blend-mode-inherit)
+                          (blend-factor-value-to-blend-mode (.m_BlendFactor ro))
+                          renderable-blend-mode)
+             face-winding (if (not= (.m_FaceWindingCCW ro) 0) GL/GL_CCW GL/GL_CW)
+             _ (set-constants! gl shader ro)
+             ro-transform (double-array (.m (.m_WorldTransform ro)))
+             renderable-transform (Matrix4d. (:world-transform renderable)) ; make a copy so we don't alter the original
+
+             ro-matrix (doto (Matrix4d. ro-transform) (.transpose))
+             shader-world-transform (doto renderable-transform (.mul ro-matrix))
+             use-index-buffer (not= (.m_UseIndexBuffer ro) 0)
+             triangle-mode (if (not= (.m_IsTriangleStrip ro) 0) GL/GL_TRIANGLE_STRIP GL/GL_TRIANGLES)
+             render-args (merge render-args
+                                (math/derive-render-transforms shader-world-transform
+                                                               (:view render-args)
+                                                               (:projection render-args)
+                                                               (:texture render-args)))]
+            (when-not is-selection-pass?
```

Comment:
use blend mode only if it's not a selection pass

Reply:
It's needed to avoid "eclipse" when a few passes of the same component are drawn on top of each other with a different "id" (color in the RT).

![image](https://github.com/user-attachments/assets/fb788448-a157-42fa-ae5d-9cc523bbb927)